code/playerman/playercontrol.cpp

```diff
@@ -916,31 +916,45 @@ void read_keyboard_controls( control_info * ci, float frame_time, physics_info *
 	}
 }
 
-void copy_control_info(control_info *dest_ci, control_info *src_ci)
+void copy_control_info(control_info *dest_ci, control_info *src_ci, int control_copy_method)
 {
-	if (dest_ci == NULL)
+	if (dest_ci == nullptr)
 		return;
 
-	if (src_ci == NULL) {
-		dest_ci->pitch = 0.0f;
-		dest_ci->vertical = 0.0f;
-		dest_ci->heading = 0.0f;
-		dest_ci->sideways = 0.0f;
-		dest_ci->bank = 0.0f;
-		dest_ci->forward = 0.0f;
-		dest_ci->forward_cruise_percent = 0.0f;
-		dest_ci->fire_countermeasure_count = 0;
-		dest_ci->fire_secondary_count = 0;
-		dest_ci->fire_primary_count = 0;
-	} else {
-		dest_ci->pitch = src_ci->pitch;
-		dest_ci->vertical = src_ci->vertical;
-		dest_ci->heading = src_ci->heading;
-		dest_ci->sideways = src_ci->sideways;
-		dest_ci->bank = src_ci->bank;
-		dest_ci->forward = src_ci->forward;
-		dest_ci->forward_cruise_percent = src_ci->forward_cruise_percent;
+	// check which type of controls are being copied --wookieejedi
+	if (control_copy_method & LGC_STEERING) {
+		if (src_ci == nullptr) {
+			dest_ci->pitch = 0.0f;
+			dest_ci->heading = 0.0f;
+			dest_ci->bank = 0.0f;
+		} else {
+			dest_ci->pitch = src_ci->pitch;
+			dest_ci->heading = src_ci->heading;
+			dest_ci->bank = src_ci->bank;
+		}
+	} else if (control_copy_method & LGC_FULL) {
```

Comment:
It looks like `control_copy_method` is a bitfield, which means it would theoretically be possible for *both* LGC_ flags to be set, wouldn't it?  In which case, the LGC_FULL check should come first.

Reply:
Good catch, just updated.